src/com/intellij/rt/coverage/util/diff/CoverageDiff.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2000-2020 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intellij.rt.coverage.util.diff;
+
+import com.intellij.rt.coverage.data.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/** This class is used for comparing coverage for tests. Checks full match of coverage results. */
+public class CoverageDiff {
```

Comment:
Do you need this testOnly code in IDEA or why do you put it in sources root?

Reply:
I hoped that it could help with https://youtrack.jetbrains.com/issue/IDEA-37125, but it does not as it does not deal with line shift. I'll move this code to the test directory for now.